src/plugins/discover/public/context_awareness/hooks/use_root_profile.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect, useState } from 'react';
+import { useDiscoverServices } from '../../hooks/use_discover_services';
+
+export const useRootProfile = ({ solutionNavId }: { solutionNavId: string | null }) => {
+  const { profilesManager } = useDiscoverServices();
+  const [rootProfileLoading, setRootProfileLoading] = useState(true);
+
+  useEffect(() => {
+    let aborted = false;
+
+    setRootProfileLoading(true);
+
+    profilesManager.resolveRootProfile({ solutionNavId }).then(() => {
```

Comment:
**question**: as we have async resolution of the profiles, should we handle errors here in case a profile resolution triggers one? Also the `resolveRootProfile` method and the other resolution functions don't encapsulate the error, would be nice to also wrap the error there to something like `ProfileResolutionError` than can be easily recognized and handled on this custom hook or where might be necessary.

Reply:
I'm good with having the error handling centralized in `ProfilesManager` as far as there are no surprising behaviours for the user when the profile resolution fails 😄

Reply:
Sounds good! Added initial error handling and fallback logic here: https://github.com/elastic/kibana/pull/183797/commits/faa68c2af38b0b174ab86bebd4625e5e72dfb12e.